palak/src/pages/Login.jsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from "react";
+import { GoEye } from "react-icons/go";
+import { GoEyeClosed } from "react-icons/go";
+import { z } from "zod";
+import { useNavigate } from "react-router-dom";
+import { Input } from "../components/Input";
+import withLoading from "../components/withLoading";
+import Button from "../components/Button";
+import { Label } from "../components/Label";
+import { loginUser } from "../api/FetchApi";
+
+const schema = z.object({
+  email: z.string().email("Must be a valid email address"),
+  password: z.string().min(6, "Password must be 6 or more characters long"),
+});
+
+const ButtonLoader = withLoading(Button);
+
+const Login = ({ handleClick }) => {
+  const [showPsswrd, setshowPsswrd] = useState(false);
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState({});
+  const [wrongError, setWrongError] = useState(false);
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const result = schema.safeParse({
+      ...formData,
+    });
+    if (!result.success) {
+      const formattedErrors = result.error.format();
+      setError(formattedErrors);
+    } else {
+      setError({});
+    }
+  };
+
+  const handleShow = () => {
+    setshowPsswrd(!showPsswrd);
+  };
+
+  const handleLogin = async () => {
+    setIsLoading(true);
+    try {
+      const data = await loginUser("auth/login", formData);
+      console.log(data);
+
+      if (data.data.token_details.token) {
+        localStorage.setItem("jwtToken", data.data.token_details.token);
+        console.log(data.data.token_details.token);
+        navigate("/front");
+      } else {
+        alert("Login unsuccessful");
+      }
+    } catch (error) {
+      wrongError = "Wrong email or password. Try again";
```

Comment:
Here wrong error is a state variable of type boolean but you are assigning it some value and that even without using the setter function.